rts/Rendering/Units/UnitDrawer.cpp

```diff
@@ -489,19 +506,44 @@ void CUnitDrawerLegacy::DrawUnitIconsScreen() const
 	sh.Enable();
 	sh.SetUniform("alphaCtrl", 0.05f, 1.0f, 0.0f, 0.0f); // GL_GREATER > 0.05
 
+
+	const float3 camDir = camera->GetDir();
+	// We are going to clamp camDir.y at -0.5, then re-normalize it. 
+	const float unitIconCameraMinY = configHandler->GetFloat("UnitIconCameraMinY");
+
+	// This is 0 if we should be using individual unit-camera distances, and 1 if we should be using a global map-camera distance
+	const float unitIconCameraFlatDistMix = std::clamp( (unitIconCameraMinY - camDir.y) * 10.0f, 0.0f, 1.0f);
```

Comment:
perhaps the `mix` function would be desired? needs `#include "System/SpringMath.h"`
```suggestion
	const float unitIconCameraFlatDistMix = ::mix(0.0f, 1.0f, camDir.y < unitIconCameraMinY);
```

otherwise, where does the `10.0f` magic come from?

Reply:
I wouldnt mix with binary, here the point was to make a 0.1 'soft' transition between individual unit-distance and global map-distance with the 10.0f multiplier based on camera Y dir.

Reply:
Should the transition softness also be configurable? Is config float bloat bad?